agent/util-scripts/pbench-tool-meister-start

```diff
@@ -447,6 +447,21 @@ class ToolDataSink(BaseServer):
                 kind = data["kind"]
                 action = data["action"]
                 status = data["status"]
+                if status != "success":
+                    tool_meister_info = data["tool_meister_info"]
+                    for host in tool_meister_info:
+                        if len(tool_meister_info[host]["failed_tools"]) == 0:
+                            continue
+                        hostname = tool_meister_info[host]["hostname"]
+                        tool = tool_meister_info[host]["installs"]
+                        for t in tool:
+                            tool_info = tool[t][1].split(": ")[1]
```

Comment:
Almost any time you iterate through the keys of a dict and then reference the *value*, you're better off with `.items()` to avoid the extra lookup; and it can make the code easier to read, as well, by eliminating the indexing (`host_info["failed_tools"]` instead of `tool_meister_info[host]["failed_tools"]`)

Finally, I think expressions like `tool[t][1].split(": ")[1]` deserve a bit of explanation about what's expected; because finding the code that constructs this payload isn't trivial and it's impossible to really analyze this without knowing what's there.

Reply:
> you're better off with `.items()`

I second this suggestion.  And, if you embrace it, then you don't need the `tool_meister_info` variable, either.

However, as I look at this, we don't actually need the keys!  So, something like this might suffice:
```
                    for entry in data["tool_meister_info"].values():
                        if len(entry["failed_tools"]) == 0:
                            continue
                        hostname = entry["hostname"]
                        tool = entry["installs"]
                        ...
```

> I think expressions like `tool[t][1].split(": ")[1]` deserve a bit of explanation

I agree.  And, I think that the `[1]` deserves to be a named reference (i.e., `tool[t]` should be a `namedtuple` or a `dict`) instead, but that's presumably beyond the scope of this PR.

Reply:
One additional suggestion is to hide these manipulations inside a reporting function, rather than have them in-line. IMO, that makes it much easier to read the code: the function name says what it is doing, but its innards can be studied later - otherwise, the reader comes to a screeching halt, going from high level considerations to extremely low-level ones.